module/zfs/zio.c

```diff
@@ -1107,9 +1107,6 @@ zio_read(zio_t *pio, spa_t *spa, const blkptr_t *bp,
 {
 	zio_t *zio;
 
-	(void) zfs_blkptr_verify(spa, bp, flags & ZIO_FLAG_CONFIG_WRITER,
-	    BLK_VERIFY_HALT);
-
```

Comment:
The arc is not the only caller of zio_read(). Is it appropriate for all other callers that we no longer verify the blkptr?

Related: the existing semantics (calling zfs_panic_recover on error) seem a bit over-the-top. Although I suppose that the logic here is that a blkptr in this context should be coming from a checksum-verified block, so we should never really encounter an invalid blkptr...

Reply:
I think it's actually preferable that we move this.  As you pointed out in `zio_read()` it's really a requirement that the block pointer be valid (there's nothing to do but panic if it isn't!), so this is really just an additional correctness check.  Moving it up in to `arc_read()` at least gives us the option to handle it as an error for this call path.  I wouldn't be against doing this same check for the other `zio_read()` callers, but it seems a little out of scope for this particular deadlock fix to me.